rest_framework_json_api/utils.py

```diff
@@ -52,7 +52,10 @@ def get_resource_name(context, expand_polymorphic_types=False):
         resource_name = getattr(view, 'resource_name')
     except AttributeError:
         try:
-            serializer = view.get_serializer_class()
+            if 'kwargs' in context and 'related_field' in context['kwargs']:
```

Comment:
@sliverc please take a look here. I'm not quite sure if there's any potential negative impact of this change.

Reply:
As outlined below I don't see that this has a negative effect as this only takes place when a related url is called.